cores/cv32e40p/user_manual/source/apu.rst

```diff
@@ -7,6 +7,9 @@ Auxiliary Processing Unit Interface
 -----------------------------------
 
 :numref:`Auxiliary Processing Unit interface signals` describes the signals of the Auxiliary Processing Unit interface.
+This interface is used to send decoded instructions to an external co-processor that receives the operands and the operation and returns the result of such operation. The CV32E40P core uses it to process RV32F instructions (see Chapter 11 of the `RISC-V Instruction Set Manual, Volume I: User-Level ISA`).
+The CV32E40P core prepares the operands (apu\_operands\_o,  and apu\_flags\_o) and the FP operation (apu\_op\_o) in the ID stage (i.e., decoding the instruction and reading the register file) and sends them to the external FPU unit in the EX stage. The FPU takes at least one cycle (depending on the instruction) to compute the operation, sent back through the apu\_result\_i and apu\_flags\_i signals. The results are then written in the register file of the core. The core is responsible for handling stalls and data-hazards.
```

Comment:
Several questions here:
- _the FPU takes at least one cycle..._.  Is this a requirement of the core?
- what is a _stall_ in this context?   How it is signalled?
- what is a _data-hazard_ in this context?
- How are stalls and data-hazards properly handled?  What happens if they are improperly handled?

Reply:
While updating the testbenches to match the changes to the APU port map (see openhwgroup/cv32e40p#558) I noticed a set of parameters whose purpose is not documented (APU_NARGS_CPU, APU,_WOP_CPU, APU_NDSFLAGS_CPU ad APU_NUSFLAGS_CPU).   Given that these are parameters of a top-level interface to an external custom device, we should capture their purpose here.